Update servicenow plugin docs to include auto approval info (#34209)
gravitational/teleport: docs/pages/access-controls/access-request-plugins/servicenow.mdx

Teleport's ServiceNow integration allows you to treat Teleport Access
Requests as ServiceNow incidents, notify the appropriate on-call team, and
approve or deny the requests via Teleport. You can also configure the plugin to
approve Access Requests automatically if the user making the request is in

> a specific on-call rotation

Reviewer: ```suggestion
a specific on-call rotation.
```

Before making the access request ensure the user making the request has
the `requester` role.

> For the plugin to know which serviceno rotations to check for the
> auto approval flow, the rota you wish users to be checked against
> must be included in a role the requesting user has. e.g.

Reviewer: ```suggestion
For the plugin to know which ServiceNow rotations to check for the
auto approval flow, the rotation you wish users to be checked against
must be included in one of the requesting user's roles. To do so, add the `teleport.dev/schedules` annotation as shown in the following example: 
```

> ```

Reviewer: ```suggestion
```yaml
```

kind: role
version: v5
metadata:
  name: YOUR_ROLE_NAME_HERE
spec:
  allow:
    request:
      roles: ['editor']
      thresholds:
        - approve: 1
          deny: 1
      annotations:
        teleport.dev/schedules:

>         - YOUR_SERVICENOW_ROTA_ID_HERE

Reviewer 2: We should explain below how to determine the rotation ID.

Author: Added the method with the fewest steps I could find.

Before making the access request ensure the user making the request has
the `requester` role.

For the plugin to know which ServiceNow rotations to check for the
auto approval flow, the rotation you wish users to be checked against
must be included in one of the requesting user's roles. To do so, add the `teleport.dev/schedules` annotation as shown in the following example: 

```yaml
kind: role
version: v5
metadata:
  name: YOUR_ROLE_NAME_HERE
spec:
  allow:
    request:
      roles: ['editor']
      thresholds:
        - approve: 1
          deny: 1
      annotations:
        teleport.dev/schedules:
        - YOUR_SERVICENOW_ROTA_ID_HERE
```

> To retrieve the ServiceNow rotation ID, first navigate to the group record
>  of the ServiceNow group the rotation belongs to and right click on
>  header->'Select copy sys_id' to copy the sys_id.

Reviewer 3: ```suggestion
To retrieve the ServiceNow rotation ID, navigate to the group record
of the ServiceNow group the rotation belongs to and right click on
header, then click 'Select copy sys_id' to copy the ID.
```

> Then using the ServiceNow endpoint '/api/now/on_call_rota/workbench/group/{groupSysId}'

Reviewer 3: This still feels hard to follow. How am I supposed to hit the endpoint? Do I need credentials?

Author: Good point. Would an example curl request be better?

In ServiceNow, you will see a new incident containing information about the
Access Request.

> If the a ServiceNow rotation was specified in the requesters role annotations,

Reviewer: ```suggestion
If the a ServiceNow rotation was specified in the requester's role annotations,
```

> then the ServiceNow plugin will check that the requesters username matches the

Reviewer: ```suggestion
then the ServiceNow plugin will check that the requester's username matches the
```
We use apostrophes in possessive nouns

> ServiceNow user currently on-call in that rotation and approve the access request.

Reviewer: ```suggestion
ServiceNow user currently on-call in that rotation and approve the Access Request.
```
For consistency with the rest of the docs.
